lib/src/migrators/namespace.dart

```diff
@@ -0,0 +1,245 @@
+// Copyright 2021 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:args/args.dart';
+import 'package:sass/sass.dart';
+import 'package:source_span/source_span.dart';
+
+// The sass package's API is not necessarily stable. It is being imported with
+// the Sass team's explicit knowledge and approval. See
+// https://github.com/sass/dart-sass/issues/236.
+import 'package:sass/src/ast/sass.dart';
+import 'package:sass/src/import_cache.dart';
+
+import '../migration_visitor.dart';
+import '../migrator.dart';
+import '../patch.dart';
+import '../exception.dart';
+import '../utils.dart';
+import '../renamer.dart';
+import 'namespace/unreferenced_flag.dart';
+
+/// Changes namespaces for `@use` rules within the file(s) being migrated.
+class NamespaceMigrator extends Migrator {
+  final name = "namespace";
+  final description = "Change namespaces for `@use` rules.";
+
+  @override
+  final argParser = ArgParser()
+    ..addMultiOption('rename',
+        abbr: 'r',
+        splitCommas: false,
+        help: 'Change namespaces based on a regular expression.\n'
+            'This can either be in the form `old-namespace to new-namespace`\n'
+            'or `url rule-url to new-namespace`, where `old-namespace and \n'
+            '`rule-url` are regular expressions and `new-namespace` can \n'
+            r'refer to captured groups with `\#`.')
+    ..addOption('unreferenced',
+        abbr: 'u',
+        help: 'Whether to change namespaces of unreferenced rules to '
+            '`_unreferenced#`.',
+        allowed: ['conflicting', 'all', 'none'],
+        allowedHelp: {
+          'conflicting': 'Only when they conflict with other rules.',
+          'all': 'All unreferenced rules will be renamed.',
+          'none':
+              'Unreferenced rules will be treated the same as referenced ones.'
+        },
+        defaultsTo: 'conflicting')
+    ..addFlag('force',
+        abbr: 'f',
+        help: 'Force rename namespaces, adding numerical suffixes for '
+            'conflicts.');
+
+  @override
+  Map<Uri, String> migrateFile(
+      ImportCache importCache, Stylesheet stylesheet, Importer importer) {
+    Renamer<UseRule> renamer;
+    try {
+      renamer = Renamer(argResults['rename'].join('\n'), 'namespace', {
+        'namespace': (rule) => rule.namespace,
+        'url': (rule) => rule.url.toString()
+      });
+    } on FormatException catch (e) {
+      throw MigrationException(e.message);
+    }
```

Comment:
Also catch a `SourceSpanFormatException` and convert it to a `MigrationSourceSpanException`.